lib/ansible/modules/cloud/amazon/aws_api_gateway.py

```diff
@@ -0,0 +1,242 @@
+    except Exception as e:
+        msg="Unexpected exception configuring AWS API connection"
+        module.fail_json(msg=msg, exception=traceback.format_exc())
+
+    if not api_id:
+        desc="Incomplete API creation by ansible aws_api_gateway module"
+        awsret=client.create_rest_api(name="ansible-temp-api", description=desc)
+        api_id=awsret["id"]
+
+    apidata=None
+    if swagger_file:
+        try:
+            with open(swagger_file) as f:
+                apidata=f.read()
+        except Exception as e:
+            module.fail_json(msg=str(e), exception=traceback.format_exc())
+    if swagger_dict:
+        apidata = json.dumps(swagger_dict)
+    if swagger_text:
+        apidata = swagger_text
+
+    if apidata is None:
+        module.fail_json(msg='module error - failed to get API data')
+
+    try:
+        create_response=client.put_rest_api(body=apidata, restApiId=api_id, mode="overwrite")
+    except botocore.exceptions.ClientError as e:
+        msg="Client side error configuring api {} - check definitions".format(api_id)
+        module.fail_json(msg=msg, exception=traceback.format_exc())
+    except botocore.exceptions.EndpointConnectionError as e:
+        msg="Connectivity problem configuring api {}. Check network.".format(api_id)
+        module.fail_json(msg=msg, exception=traceback.format_exc())
+    except botocore.exceptions.NoCredentialsError as e:
+        msg="AWS credentials missing configuring api {}. set up credentials.".format(api_id)
+        module.fail_json(msg=msg, exception=traceback.format_exc())
+    except Exception as e:
+        msg="Unexpected exception configuring api {}".format(api_id)
+        module.fail_json(msg=msg, exception=traceback.format_exc())
+
+    if deploy_desc is None:
+        deploy_desc = "Automatic deployment by Ansible."
+    if stage:
+        try:
+            deploy_response=client.create_deployment(restApiId=api_id, stageName=stage,
+                                                     description=deploy_desc)
+        except (botocore.exceptions.ClientError) as e:
+            msg="Client side error deploying api {} to stage {}".format(api_id, stage)
+            module.fail_json(msg=msg, exception=traceback.format_exc())
+        except botocore.exceptions.EndpointConnectionError as e:
+            msg="Connectivity problem deploying api {} to stage {}".format(api_id, stage)
+            module.fail_json(msg=msg, exception=traceback.format_exc())
+        except Exception as e:
+            msg="Unexpected exception deploying api {} to stage {}".format(api_id, stage)
+            module.fail_json(msg=msg, exception=traceback.format_exc())
+    changed=True
+    module.exit_json(changed=changed, api_id=api_id,
+                     create_response=camel_dict_to_snake_dict(create_response),
+                     deploy_response=camel_dict_to_snake_dict(deploy_response))
+
+from ansible.module_utils.basic import *
```

Comment:
module imports should be explicit and near the top of the file just below documentation

See e.g. https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/ec2_vpc_nat_gateway_facts.py#L92-L94